ports/espressif/common-hal/watchdog/WatchDogTimer.c

```diff
@@ -44,26 +56,36 @@ void esp_task_wdt_isr_user_handler(void) {
 }
 
 void common_hal_watchdog_feed(watchdog_watchdogtimer_obj_t *self) {
-    if (esp_task_wdt_reset() != ESP_OK) {
-        mp_raise_RuntimeError(translate("watchdog not initialized"));
-    }
+    esp_task_wdt_reset();
 }
 
 void common_hal_watchdog_deinit(watchdog_watchdogtimer_obj_t *self) {
+    if (self->mode == WATCHDOGMODE_NONE) {
+        return;
+    }
     if (esp_task_wdt_delete(NULL) == ESP_OK && esp_task_wdt_deinit() == ESP_OK) {
         self->mode = WATCHDOGMODE_NONE;
     }
 }
 
 void watchdog_reset(void) {
-    common_hal_watchdog_deinit(&common_hal_mcu_watchdogtimer_obj);
+    watchdog_watchdogtimer_obj_t *self = &common_hal_mcu_watchdogtimer_obj;
+    if (self->mode == WATCHDOGMODE_RESET) {
+        mp_obj_t exception = pyexec_result()->exception;
+        if (exception != MP_OBJ_NULL &&
+            exception != MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_kbd_exception)) &&
+            exception != MP_OBJ_FROM_PTR(&MP_STATE_VM(mp_reload_exception))) {
```

Comment:
I feel like it's unfortunate that this condition is repeated multiple times in multiple ports. can you introduce a function such as `shared_module_watchdog_should_deinit_on_reset` and call it?